src/components/views/dialogs/HostingSignupDialog.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as React from "react";
+import * as sdk from '../../../index';
+import Modal from "../../../Modal";
+import SdkConfig from "../../../SdkConfig";
+import { _t } from "../../../languageHandler";
+import { MatrixClientPeg } from "../../../MatrixClientPeg";
+
+interface IProps {
+    requestClose(): void,
+}
+
+interface IState {
+    completed: boolean,
+    error: string,
+}
+
+export default class HostingSignupDialog extends React.PureComponent<IProps, IState> {
+    iframeRef;
+    hostingSignupUrl: string;
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            completed: false,
+            error: null,
+        };
+
+        this.iframeRef = React.createRef();
+        this.hostingSignupUrl = SdkConfig.get().hosting_signup.url;
+    }
+
+    private messageHandler = (message) => {
+        if (!this.hostingSignupUrl.startsWith(message.origin)) {
+            return;
+        }
+        switch (message.data.action) {
+            case 'account_credentials_request':
+                // noinspection JSIgnoredPromiseFromCall
+                this.sendAccountDetails();
+                break;
+            case 'setup_complete':
+                // Set as completed but let the user close the modal themselves
+                // so they have time to finish reading any information
+                this.setState({
+                    completed: true,
+                });
+                break;
+            case 'openid_credentials_request':
+                // noinspection JSIgnoredPromiseFromCall
+                this.fetchOpenIDToken();
+                break;
```

Comment:
why are we not prompting the user on this?

Reply:
Will refactor into a widget.

Reply:
The decision was made not to use a widget after all, as we need to query for the access token for this feature to work. This is a feature we do not want to have for widgets.